File: my-messenger-backend-java/src/main/java/mymessenger/backend/ws/StreamMessageResponse.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package mymessenger.backend.ws;

import mymessenger.backend.model.messaging.Message;
import mymessenger.backend.model.users.UserSession;

/**
 *
 * @author guilherme
 */
public class StreamMessageResponse {

    public static StreamMessageResponse ok() {
        StreamMessageResponse r = new StreamMessageResponse();
        r.setOk(true);
        return r;
    }

    public static StreamMessageResponse userSession(UserSession us) {
        StreamMessageResponse r = new StreamMessageResponse();
        r.setOk(true);
        r.setUserSession(us);
        return r;
    }

    static StreamMessageResponse message(Message msg) {
        StreamMessageResponse r = new StreamMessageResponse();
        r.setOk(true);
        r.setMessage(msg);
        return r;
    }
    
    private boolean ok;
    private UserSession userSession;
    private Message message;

    public boolean isOk() {
        return ok;
    }

    public void setOk(boolean ok) {
        this.ok = ok;
    }

    public UserSession getUserSession() {
        return userSession;
    }

    public void setUserSession(UserSession userSession) {
        this.userSession = userSession;
    }

    public Message getMessage() {
        return message;
    }

    public void setMessage(Message message) {
        this.message = message;
    }

}

File: my-messenger-backend/my.messenger.backend/WS/StreamMessageResponse.cs
using my.messenger.common.Messaging;
using my.messenger.common.Users;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace my.messenger.backend.WS
{
    public class StreamMessageResponse
    {
        public static StreamMessageResponse ofOk()
        {
            StreamMessageResponse r = new StreamMessageResponse();
            r.Ok = (true);
            return r;
        }

        public static StreamMessageResponse ofUserSession(UserSession us)
        {
            StreamMessageResponse r = new StreamMessageResponse();
            r.Ok = (true);
            r.UserSession = (us);
            return r;
        }

        public static StreamMessageResponse ofMessage(Message msg)
        {
            StreamMessageResponse r = new StreamMessageResponse();
            r.Ok = (true);
            r.Message = (msg);
            return r;
        }

        public bool Ok { get; set; }
        public UserSession UserSession { get; set; }
        public Message Message { get; set; }
    }
}
